internal/test/resources.go

```diff
@@ -2150,38 +2216,39 @@ func OtherServiceAccount() *corev1.ServiceAccount {
 }
 
 func NewRole() *rbacv1.Role {
+	rules := []rbacv1.PolicyRule{
```

Comment:
In case built-in discovery is disabled, I believe we can remove these rules (or at least `endpoints`. But I am a bit unsure if these rules use cases are beyond built-in discovery mechanism. Right now, I don't remove any, but any suggestions on these?

Reply:
True, I think `endpoints` is only used for the old-style discovery system. If the new `BuiltInDiscoveryDisabled` property is `true` then there should be nothing making k8s API requests for that specific resource type.

The rest look like permissions that the Operator needs to be able to set up the deployments and routes for things.

I don't remember exactly - are there two separate service accounts for the Operator and for Cryostat itself? Cryostat would only need `endpoints` from this list, and the Operator would need everything but `endpoints` I think (which used to be used for the FlightRecorder CR).

Reply:
Right, there are 2 separate SA for operator and Cryostat. I think the operator still needs `endpoints` access in order to gain that to Cryostat SA. Do you think these permissions are somewhat used for Automated Rules?

Reply:
Automated Rules goes through the same target discovery code as everything else within Cryostat, so Automated Rules only needs Endpoints indirectly if the KubeApiPlatformClient is in use (via the BuiltInDiscovery hooking it into the DiscoveryStorage). If the property/envvar is set and built-in discovery is disabled then Automated Rules, and other things, will continue just querying the DiscoveryStorage for what it does know about, and in this scenario it would not be populated by KubeApiPlatformClient so no Endpoints queries would ever be made.

Reply:
It makes sense that the Operator might need to have a strict superset of the permissions it might ever grant to Cryostat. But in the case where this property is set and disables built-in discovery, the Cryostat SA shouldn't need that permission anymore and it should probably not be granted.

Reply:
Right, make sense to remove those accesses if not needed. I can remove the `endpoints` if disabling built-in mechanism. Just a bit unsure what are other permissions used for so I can remove them too. Any lead for this too @ebaron?

Reply:
Right...For now (or this PR), I think we can avoid modifying roles :D